jina/orchestrate/flow/base.py

```diff
@@ -2353,24 +2366,94 @@ def block(
+            self.logger.info(
+                f'change in Executor configuration YAML {changed_file} observed, restarting Executor deployment'
+            )
+            deployment.__exit__(None, None, None)
+            old_args, old_needs = deployment.args, deployment.needs
+            new_deployment = Deployment(old_args, old_needs)
+            deployment.__dict__ = new_deployment.__dict__
+            deployment.__enter__()
+
         try:
-            if stop_event is None:
-                self._stop_event = (
-                    threading.Event()
-                )  #: this allows `.close` to close the Flow from another thread/proc
-                if not __windows__:
-                    self._stop_event.wait()
-                else:
-                    while True:
-                        if self._stop_event.is_set():
-                            break
-                        time.sleep(0.5)
+            watch_changes = self.args.restart or any(
+                [
+                    deployment.args.restart
+                    for deployment in list(self._deployment_nodes.values())
+                ]
+            )
+            watch_files_from_deployments = {}
+            for name, deployment in self._deployment_nodes.items():
+                if deployment.args.restart:
+                    if deployment._is_executor_from_yaml:
+                        watch_files_from_deployments[deployment.args.uses] = name
+            watch_files_list = list(watch_files_from_deployments.keys())
+
+            config_loaded = getattr(self, '_config_loaded', '')
+            if config_loaded.endswith('yml') or config_loaded.endswith('yaml'):
+                watch_files_list.append(config_loaded)
+
+            if watch_changes and len(watch_files_list) > 0:
+
+                with ImportExtensions(
+                    required=True,
+                    logger=self.logger,
+                    help_text='''restart requires watchfiles dependency to be installed. You can do `pip install 
+                    watchfiles''',
+                ):
+                    from watchfiles import watch
+
+                new_stop_event = stop_event or threading.Event()
+                if len(watch_files_list) > 0:
+                    for changes in watch(
+                        *watch_files_list, stop_event=new_stop_event
+                    ):
+                        for _, changed_file in changes:
+                            if changed_file not in watch_files_from_deployments:
+                                # maybe changed_file is the absolute path of one in watch_files_from_deployments
+                                is_absolute_path = False
+                                for (
+                                    file,
+                                    deployment_name,
+                                ) in watch_files_from_deployments.items():
+                                    if changed_file.endswith(file):
```

Comment:
Can this be flattened somehow? There are quite some levels of indentation here

Reply:
No, it is not possible I'd say